### PR TITLE
tests: Fix thread return with local message queue

### PR DESCRIPTION
**Pull request by chrysn**
### Contribution description

When a message queue is configured from the stack, that main function
must never return -- otherwise, during sched_task_exit (which the
thread's function "returns" to), message senders might still send
messages into already freed stack space (which would be reused by
sched_task_exit).

[new] A follow-up commit adds a line on this to the msg_init_queue documentation.

Most occurrences of queue initialization throughout RIOT are either static or followed by an endless loop without breaks / returns; the few examples not matching either pattern are fixed here.

### Testing procedure

I don't see any way to make the fixed behavior actually happen; this is more about best practice. (Especially given that sched_task_exit currently starts with a debug statement followed by `irq_disable`, so the message would need to hit right into the execution of debug printing).

The two altered tesets that don't depend on hardware can be and were successfully run (in tests/thread_msg_block_w_queue and tests/posix_semaphore) on native.

### Issues/PRs references

This came up when nailing down the safety criteria of `msg_init_queue` in Rust.

[edit: see [new]]

### CODE REVIEW

File: tests/driver_nrf24l01p_lowlevel/main.c [hunk 1]
```diff
@@ -109,12 +109,13 @@ void print_register(char reg, int num_bytes)
 
 char rx_handler_stack[THREAD_STACKSIZE_MAIN];
 
+msg_t _msg_q[1];
```
Comment (maribu) [inline tests/driver_nrf24l01p_lowlevel/main.c:112]:
```suggestion
static msg_t _msg_q[1];
```

File: tests/posix_semaphore/main.c [hunk 2]
```diff
@@ -129,10 +129,10 @@ static void test1(void)
     puts("first: end");
 }
 
+msg_t _sema_thread_queue[SEMAPHORE_MSG_QUEUE_SIZE];
```
Comment (maribu) [inline tests/posix_semaphore/main.c:132]:
```suggestion
static msg_t _sema_thread_queue[SEMAPHORE_MSG_QUEUE_SIZE];

```

File: tests/posix_semaphore/main.c [hunk 3]
```diff
@@ -176,23 +176,23 @@ void test2(void)
     }
 }
 
+msg_t _one_two_queue[SEMAPHORE_MSG_QUEUE_SIZE];
```
Comment (maribu) [inline tests/posix_semaphore/main.c:179]:
```suggestion
static msg_t _one_two_queue[SEMAPHORE_MSG_QUEUE_SIZE];

```

File: tests/posix_semaphore/main.c [hunk 4]
```diff
@@ -176,23 +176,23 @@ void test2(void)
...
-    msg_init_queue(msg_queue, SEMAPHORE_MSG_QUEUE_SIZE);
+    msg_init_queue(_one_two_queue, SEMAPHORE_MSG_QUEUE_SIZE);
     sem_wait(&s1);
     puts("Thread 1 woke up after waiting for s1.");
     sem_wait(&s2);
     puts("Thread 1 woke up after waiting for s2.");
     return NULL;
 }
 
+msg_t _two_one_queue[SEMAPHORE_MSG_QUEUE_SIZE];
```
Comment (maribu) [inline tests/posix_semaphore/main.c:191]:
```suggestion
static msg_t _two_one_queue[SEMAPHORE_MSG_QUEUE_SIZE];

```

File: tests/thread_msg_block_w_queue/main.c [hunk 5]
```diff
@@ -52,13 +52,13 @@ void *sender_thread(void *arg)
     return NULL;
 }
 
+msg_t _msg_q[1];
```
Comment (maribu) [inline tests/thread_msg_block_w_queue/main.c:55]:
```suggestion
static msg_t _msg_q[1];

```

**Review by maribu — COMMENTED** [review]
Looks good to me. I think the arrays could be `static`, couldn't they?

(I'm aware it is style and subjective, but a blank line before a function makes it IMO easier to spot where the function starts. Feel free to ignore, if you disagree.)

**Comment by chrysn** [comment]
Yes they can be static, and probably should (not that it'd matter a lot in the test applications); going with your newlines as well.

**Review by maribu — APPROVED** [review]
ACK. The change is unscary enough to go in without testing. Please squash :)

**Comment by maribu** [comment]
No need for a second ACK IMO. If you insist on a 2nd ACK, speak up now.

**Pull request merged**